### ISSUE TITLE
sourceName not replaced in Folder/File names in 2.0-preview2

### ISSUE THREAD

**Issue by MarcinJuraszek** [dotnet/templating#1181]
When switching from dotnet SDK 1.4 to dotnet SDK 2.0 preview 2 I noticed a change in how dotnet templating works and I'd like to know if that's expected (basically an intentional breaking change) or a bug.

With 1.4 if `sourceName` was present in any folder or file name it would be replaced. SDK 2.0 preview 2 does not perform replacement in folder/file names.

For example, if the template.json contained `sourceName: "HelloWorld"` and following folder structure 

```
└───HelloWorld
    │   dirs.proj
    │   HelloWorld.sln
    │   HelloWorld.v3.ncrunchsolution
    │
    ├───HelloWorld
    │       Hello.cs
    │       HelloWorld.csproj
    │       HelloWorld.v3.ncrunchproject
    │       Initializer.cs
    │
    ├───HelloWorld.Exports
    │       HelloWorld.Exports.csproj
    │       HelloWorld.Exports.v3.ncrunchproject
    │       IHello.cs
    │
    ├───HelloWorld.Extensions
    │       HelloExtensions.cs
    │       HelloWorld.Extensions.csproj
    │       HelloWorld.Extensions.v3.ncrunchproject
    │
    └───HelloWorld.UnitTests
            HelloWorld.UnitTests.csproj
            HelloWorld.UnitTests.v3.ncrunchproject
            HelloWorldTest.cs
```

after running `dotnet new lib -n MyNewLibrary`  with SDK 1.4 I get following template created:

```
└───MyNewLibrary
    │   dirs.proj
    │   MyNewLibrary.sln
    │   MyNewLibrary.v3.ncrunchsolution
    │
    ├───MyNewLibrary
    │       Hello.cs
    │       Initializer.cs
    │       MyNewLibrary.csproj
    │       MyNewLibrary.v3.ncrunchproject
    │
    ├───MyNewLibrary.Exports
    │       IHello.cs
    │       MyNewLibrary.Exports.csproj
    │       MyNewLibrary.Exports.v3.ncrunchproject
    │
    ├───MyNewLibrary.Extensions
    │       HelloExtensions.cs
    │       MyNewLibrary.Extensions.csproj
    │       MyNewLibrary.Extensions.v3.ncrunchproject
    │
    └───MyNewLibrary.UnitTests
            MyNewLibrary.UnitTests.csproj
            MyNewLibrary.UnitTests.v3.ncrunchproject
            MyNewLibraryTest.cs
```

with the very same template and dotnet SDK 2.0 I'm getting following result:

```
├───HelloWorld
│   │   dirs.proj
│   │   HelloWorld.sln
│   │   HelloWorld.v3.ncrunchsolution
│   │
│   ├───HelloWorld
│   │       Hello.cs
│   │       HelloWorld.csproj
│   │       HelloWorld.v3.ncrunchproject
│   │       Initializer.cs
│   │
│   ├───HelloWorld.Exports
│   │       HelloWorld.Exports.csproj
│   │       HelloWorld.Exports.v3.ncrunchproject
│   │       IHello.cs
│   │
│   ├───HelloWorld.Extensions
│   │       HelloExtensions.cs
│   │       HelloWorld.Extensions.csproj
│   │       HelloWorld.Extensions.v3.ncrunchproject
│   │
│   └───HelloWorld.UnitTests
│           HelloWorld.UnitTests.csproj
│           HelloWorld.UnitTests.v3.ncrunchproject
│           HelloWorldTest.cs
```

cc: @rachit1503

**Comment by seancpeters** [comment]
Hi @MarcinJuraszek - the file rename behavior in 1.4 is correct and is the intended behavior in 2.0 preview 2. To help troubleshoot this issue, please copy-paste to here the output from running `dotnet --version`. Also, if you're able to provide an install link to your template, that'd be great. If not, could you copy-paste the contents of your template.json here?

**Comment by MarcinJuraszek** [comment]
`dotnet --version` returned `2.0.0-preview2-006497`. 

Here's the template.json:

```
{
  "identity": "Microsoft.Office.Online.Templates.Library",
  "author": "marcinj",
  "classifications": [ "Office Online", "library" ],
  "generatorVersions": "[1.0.0.0-*)",
  "name": "Office Online Library",
  "shortName": "olib",
  "tags": {
    "language": "C#"
  },
  "sourceName": "HelloWorld",
  "sources": [
    {
      "source": "./",
      "target": "./src/libs/"
    }
  ],
  "guids": [
    "FAE04EC0-301F-11D3-BF4B-00C04F79EFBC",
    "433957DA-B708-48B3-B63A-4C1BAD4F637B",
    "C7C1D0EF-7394-4311-89AB-7F0089B67EF4",
    "4FA8D1E5-23BF-4F09-848C-12A2EA634946",
    "04AA1EEB-C6E5-43C0-B0B9-73EB421D4FF8"
  ]
}
```

**Comment by seancpeters** [comment]
I'm able to repro your issue, looking into the exact cause now. I'll update this thread when I have a PR with the bug fix.

**Comment by MarcinJuraszek** [comment]
OK. That means it's not me doing something really stupid :) Looking forward to a fix, and we're going to stick with 1.4 for now.

Also, on a side note, that seems like a core scenario, so I'd expect some test fail when the functionality was broken. Maybe some tests should be added now to prevent another regression in the future?

**Comment by sayedihashimi** [comment]
@seancpeters is there a workaround that can be applied?

**Comment by seancpeters** [comment]
You're absolutely correct, I plan on adding some tests for the specific circumstances that cause the issue. It's related to the target of the source not being just "./"

**Comment by seancpeters** [comment]
For a workaround, the files in the template could be restructured so they're under the "./src/libs/" directory in the source, and changing the target to be "./". And in this specific case, since the custom source only specifies the source & target directories, it could be removed entirely, as those are the defaults.

**Comment by MarcinJuraszek** [comment]
The template is distributed in an internal NuGet feed, so I can't really move them under `./src/libs`. But the real workaround is to use 1.4 and not preview builds :)

**Comment by seancpeters** [comment]
@MarcinJuraszek This PR https://github.com/dotnet/templating/pull/1184 should fix the problem in this thread. Once it's merged, it'd be great if you could test it out against your template. In addition to the unit test in the PR, I was testing with your template,json and a mocked-up set of files / directories, but it's always good to get another test.

Thanks for pointing out this problem!

**Comment by MarcinJuraszek** [comment]
Sure, just ping me when it's merged. I assume I will have to build it from sources to get the latest?

**Comment by seancpeters** [comment]
If you pull the branch from here, first: run `setup.cmd` (or `setup.sh` on *nix), which are both in the repo base dir. This creates the dotnet command `new3`. Then you can use `dotnet new3` just like `dotnet new`, except new3 is the bits from running setup, and new is not modified. But new & new3 do not share configuration between themselves, so you'd also need to install your template in new3.

**Comment by seancpeters** [comment]
Minor update: I closed the original PR, recreated it as https://github.com/dotnet/templating/pull/1187 against a different branch.

**Comment by seancpeters** [comment]
@MarcinJuraszek - the fix for this issue is available in the newest dotnet/cli release-2.0.0 daily build, it's version 2.0.1-servicing-006933. You can download it here: https://github.com/dotnet/cli/tree/release/2.0.0

Please let us know if you're still encountering problems.
